File: go/1/main.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	hash := make(map[int]int)
	for idx, num := range nums {
		jdx, ok := hash[num]
		if ok {
			return []int{idx, jdx}
		}

		hash[target-num] = idx
	}

	return []int{}
}

func main() {
	fmt.Println("case 1")
	nums := []int{2, 7, 11, 15}
	target := 9
	fmt.Print("result: ", twoSum(nums, target))

	fmt.Println()

	fmt.Println("case 2")
	nums = []int{3, 2, 4}
	target = 6
	fmt.Print("result: ", twoSum(nums, target))

	fmt.Println()

	fmt.Println("case 3")
	nums = []int{3, 3}
	target = 6
	fmt.Print("result: ", twoSum(nums, target))
	
	fmt.Println()
}

File: python/1/main.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}

        for i in range(len(nums)):
            if nums[i] in hash:
                return [i, hash[nums[i]]]

            hash[target-nums[i]] = i

        return []
    

solution = Solution()

# case 1
result = solution.twoSum([2,7,11,15], 9)
print(result)